fabric-model-loading-api-v1/src/client/java/net/fabricmc/fabric/api/client/model/loading/v1/SimpleExtraModel.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.model.loading.v1;
+
+import java.util.function.BiFunction;
+
+import net.minecraft.client.render.model.BakedSimpleModel;
+import net.minecraft.client.render.model.Baker;
+import net.minecraft.client.render.model.BlockStateModel;
+import net.minecraft.client.render.model.GeometryBakedModel;
+import net.minecraft.client.render.model.ModelBakeSettings;
+import net.minecraft.client.render.model.ModelRotation;
+import net.minecraft.client.render.model.ModelTextures;
+import net.minecraft.client.render.model.SimpleBlockStateModel;
+import net.minecraft.util.Identifier;
+
+/**
+ * A {@link UnbakedExtraModel} that loads a single model.
+ *
+ * @param <T> The type of the baked model, for instance {@link BlockStateModel}.
+ */
+public final class SimpleExtraModel<T> implements UnbakedExtraModel<T> {
+	private final Identifier model;
+	private final BiFunction<BakedSimpleModel, Baker, T> bake;
+
+	/**
+	 * @param model The location of the model to load.
+	 * @param bake  A function to bake the model.
+	 */
+	public SimpleExtraModel(Identifier model, BiFunction<BakedSimpleModel, Baker, T> bake) {
+		this.model = model;
+		this.bake = bake;
+	}
+
+	/**
+	 * Create a {@link SimpleExtraModel} for a {@link BlockStateModel}.
+	 *
+	 * <h2>Example</h2>
+	 * <pre>{@code
```

Comment:
This could also be a `{@snippet}`, but I doubt it matters.

Reply:
Ahh, good shout. I'd stopped using `@snippet` with Checkstyle a while back, but I think as long as you avoid using Java annotations in the codeblock it's fine.

Reply:
Yeah, it's already used in a few places in FAPI for examples (with or without `@link` spam).